src/Agent.Worker/TestResults/TestRunDataPublisherHelper.cs

```diff
@@ -0,0 +1,170 @@
+﻿using Microsoft.TeamFoundation.TestClient.PublishTestResults;
+using Microsoft.TeamFoundation.TestManagement.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Worker.TestResults.Utils;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using LegacyTestRunPublisher = Microsoft.VisualStudio.Services.Agent.Worker.LegacyTestResults.ITestRunPublisher;
+using ITestResultsServer = Microsoft.VisualStudio.Services.Agent.Worker.LegacyTestResults.ITestResultsServer;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.TestResults
+{
+    public class TestRunDataPublisherHelper
+    {
+        private IExecutionContext _executionContext;
+        private ITestRunPublisher _libraryTestRunPublisher;
+        private LegacyTestRunPublisher _agentTestRunPublisher;
+        private ITestResultsServer _testResultsServer;
+
+        public TestRunDataPublisherHelper(IExecutionContext executionContext, ITestRunPublisher libraryTestRunPublisher, LegacyTestRunPublisher agentTestRunPublisher, ITestResultsServer testResultServer)
+        {
+            _executionContext = executionContext;
+            _libraryTestRunPublisher = libraryTestRunPublisher;
+            _agentTestRunPublisher = agentTestRunPublisher;
+            _testResultsServer = testResultServer;
+        }
+
+        protected internal virtual bool? CheckRunsForFlaky(IList<TestRun> runs, string projectName)
+        {
+            bool? runOutcome = null;
+            try
+            {
+                using (var connection = WorkerUtilities.GetVssConnection(_executionContext))
+                {
+                    var featureFlagService = _executionContext.GetHostContext().GetService<IFeatureFlagService>();
+                    featureFlagService.InitializeFeatureService(_executionContext, connection);
+                    var doUseStateAPI = featureFlagService.GetFeatureFlagState(TestResultsConstants.UseStatAPIFeatureFlag, TestResultsConstants.TFSServiceInstanceGuid);
+
+                    // If null is returned then fallback to previous value.
+                    runOutcome = DoesRunsContainsFailures(runs, projectName);
+                }
+
+                return runOutcome;
+            }
+            catch (Exception ex)
+            {
+                // Exception in checking flaky will not fail the mainline pipeline
+                _executionContext.Warning("Failed to Check for Flaky : " + ex.ToString());
```

Comment:
you want to print this as Info or warning this can cause builds to fail, if they treat warnings as error.

Reply:
@navin22 Yeah we do not want to fail the pipeline due to flaky check, changed to info.